modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/exec/rel/AbstractExecutionTest.java

```diff
@@ -90,6 +116,66 @@ private void handle(Thread t, Throwable ex) {
         return fields;
     }
 
+    /** Task reordering executor. */
+    private static class IgniteTestStripedThreadPoolExecutor extends StripedThreadPoolExecutor {
+        final Deque<Pair<Runnable, Integer>> tasks = new ArrayDeque<>();
+
+        /** Internal stop flag. */
+        AtomicBoolean stop = new AtomicBoolean();
+
+        /** Inner execution service. */
+        ExecutorService exec = Executors.newWorkStealingPool();
+
+        /** {@inheritDoc} */
+        public IgniteTestStripedThreadPoolExecutor(
+                int concurrentLvl,
+                String threadNamePrefix,
+                Thread.UncaughtExceptionHandler exHnd,
+                boolean allowCoreThreadTimeOut,
+                long keepAliveTime
+        ) {
+            super(concurrentLvl, threadNamePrefix, exHnd, allowCoreThreadTimeOut, keepAliveTime);
+
+            IgniteTestUtils.runAsync(() -> {
+                while (!stop.get()) {
+                    synchronized (tasks) {
+                        while (!tasks.isEmpty()) {
+                            Pair<Runnable, Integer> r = tasks.pollFirst();
+
+                            exec.execute(() -> {
+                                LockSupport.parkNanos(ThreadLocalRandom.current().nextLong(0, 10_000));
```

Comment:
I don't think it's a good idea to introduce random delays up to 10 seconds to all the tests. For several classes this increases an execution time on order of magnitude.

Reply:
oh, looks like I was wrong, the time unit is nanos, so you park it to only 10 mills... Nevertheless, with this patch, the `ExecutionTest` runs 10 times longer on my laptop. Let's try to figure it out

Reply:
i fix a bit, now execution tests runs 1 min, is it ok?